plugins/cloud-aws/src/main/java/org/elasticsearch/cloud/aws/blobstore/S3BlobStore.java

```diff
@@ -71,13 +71,28 @@ public S3BlobStore(Settings settings, AmazonS3 client, String bucket, @Nullable
         }
 
         this.numberOfRetries = maxRetries;
-        if (!client.doesBucketExist(bucket)) {
-            if (region != null) {
-                client.createBucket(bucket, region);
-            } else {
-                client.createBucket(bucket);
+
+        int retry = 0;
+        while (retry <= this.numberOfRetries) {
```

Comment:
And, because we need to do logging after all 3 times retries failed, doing loop in this way looks OK for me.